pkg/asset/installconfig/azure/metadata.go

```diff
@@ -0,0 +1,71 @@
+package azure
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+
+	typesazure "github.com/openshift/installer/pkg/types/azure"
+)
+
+// Metadata holds additional metadata for InstallConfig resources that
+// does not need to be user-supplied (e.g. because it can be retrieved
+// from external APIs).
+type Metadata struct {
+	session   *Session
+	client    *Client
+	dnsCfg    *DNSConfig
+	CloudName typesazure.CloudEnvironment
```

Comment:
Is it worth putting this behind a getter to show that the value is immutable?  Or do we trust consumers to play nice?  Whatever the public API is, it should have a good comment.

Reply:
Hmm, I don't actually see a consumer on a quick skim.  Maybe we can use `cloudName` with no getter?

Reply:
> Is it worth putting this behind a getter to show that the value is immutable? Or do we trust consumers to play nice?

Also I think this does not need to be exported. CloudName should be available in the platform